py-rattler/src/index.rs

```diff
@@ -0,0 +1,16 @@
+use pyo3::{pyfunction, PyResult};
+use rattler_conda_types::Platform;
+use rattler_index::index;
+
+use std::path::PathBuf;
+
+use crate::{error::PyRattlerError, platform::PyPlatform};
+
+#[pyfunction]
+pub fn py_index(channel_directory: PathBuf, target_platform: Option<PyPlatform>) -> PyResult<bool> {
```

Comment:
You might be able to release the GIL here which makes it easier to run this function on a thread.

Reply:
Okay yeah, I added allow_threads, lmk if that's not what you meant